src/Angor/Client/Pages/Create.razor

```diff
@@ -578,45 +704,49 @@
 
         if (string.IsNullOrEmpty(project.Metadata.Name))
         {
-            notificationComponent.ShowErrorMessage("Nostr profile name be filled");
+            notificationComponent.ShowErrorMessage("Nostr profile name must be filled");
             return;
         }
 
-        createProfileSpinner = true;
-        StateHasChanged();
-        await Task.Delay(10);
+        confirmationMessage = "Are you sure you want to finalize the Nostr Profile creation? Once done, changes can only be made on Nostr.";
+        finalizeAction = FinalizeNostrProfileCreation;
+        showConfirmationModal = true;
+    }
 
-        try
+    private async void ConfirmFinalizeProjectInfoCreation()
+    {
+        if (!await ProjectInfoMetaDataTestInput())
```

Comment:
This should obviously be called before we publish to Nostr probably in the method `CreatNostrProfile` or even in `FinalizeNostrProfileCreation` otherwise we will post the project info to Nostr and then find it fails validation

Reply:
how is it possible we will post not validated project info?
Right now the order is after we create nostr profile, in the method we:
1) validate the inputs `if (!await ProjectInfoMetaDataTestInput())
        {
            // If validation fails, return early
            return;
        }`
        
        
2) post it to the relay with AddProjectAsync

(or when you say validation you mean other type of validation?)

Reply:
My point is we must do all validation before we create the Nostr profile, if not it is too confusing to the user imo
Imagine the metadata is not valid so they create a profile then have to go back to fix the metadata? it is also harder to check that logic programmatically.

Reply:
so basically I suggest make a new method something like this before creating the profile

```
 private async void Validate()
 {
     if (!await ValidateProfile())
     {
         // If validation fails return
         return;
     }

     if (!await ValidateMetadata())
     {
         // If validation fails return
         return;
     }
    
   // continue execution to create profile
 }
```

Reply:
do you mean missing name? or some of the metadata?
we have some checks in ConfirmFinalizeNostrProfileCreation before the pop up of creating

Reply:
I mean the point in time for the validation of `ConfirmFinalizeNostrProfileCreation` is too late, it happens after the profile was already created on nostr.